Braille graphs

squidbeam opened the pull request:
I added options to cpu_usage_graph and network to draw braille graphs.

I did this by modifying make_graph to take a "style" parameter ("blocks" for yours, "braille" for braille), and then adding the above two corresponding options.

Your blocks have greater vertical resolution, but the braille has greater horizontal resolution.
![braille](https://cloud.githubusercontent.com/assets/10637890/5926369/9b535678-a638-11e4-814d-b4ba0725d053.png)

To save you some time considering this, I attached an image (with duplicated cpu plots for comparison), and here are the configuration lines I used:
            graph_style = 'braille',
            graph_width = 32,

facetoe commented:
Nice! Do you think it's possible to make a line graph with these characters?

squidbeam commented:
Yes, no problem.

I added two new choices, one that just draws the peak, and one that tries to fill things in a little.  Please see attached image.
![braille2](https://cloud.githubusercontent.com/assets/10637890/5947933/d7ec4ae6-a70e-11e4-8a88-b0d87714c1ca.png)

facetoe commented:
Awesome! Yes if you want to add a lower limit to `make_graph` go right ahead. There is no harm in it if you set the default value to something sane.

squidbeam commented:
Great, I've added a commit with lower_limit.

There was an obvious default value of 0.0, which was already implicitly
used in cpu/network graphs when choosing the color.  I have additionally
added "None" as limit option to do the rescaling as before.  I did not
expose lower_limit in the cpu/network graph code since the default felt
fine and it seemed a bit much.

Thanks for suggesting line drawing: I personally like the
'braille-snake' graphs the most.

On Wed, Jan 28, 2015 at 04:25:41PM -0800, facetoe wrote:
> Awesome! Yes if you want to add a lower limit to `make_graph` go right ahead. There is no harm in it if you set the default value to something sane. 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/enkore/i3pystatus/pull/167#issuecomment-71945905
